tests/protractor/e2e/registration-by-sms.js

```diff
@@ -0,0 +1,279 @@
+          params: 'lmp_date',
+          bool_expr: 'doc.fields.last_menstrual_period'
+        },
+        {
+          name: 'on_create',
+          trigger: 'assign_schedule',
+          params: 'ANC Reminders LMP',
+          bool_expr: 'doc.fields.last_menstrual_period'
+        }
+      ],
+      validations: {
+        join_responses: false,
+        list: [{
+          property: 'patient_name',
+          rule: 'lenMin(1) && lenMax(30)',
+          message: [{
+            content: '{{#patient_name}}The registration format is incorrect, ensure the message starts with R followed by space and the mother name (maximum of 30 characters).{{/patient_name}}{{^patient_name}}The registration format is incorrect. ensure the message starts with R followed by space and the mothers name.{{/patient_name}}.',
+            locale: 'en'
+          }]
+        }]
+      },
+      messages: [{
+        message: [{
+          content: 'Thank you {{contact.name}} for registering {{patient_name}}',
+          locale: 'en'
+        }],
+        recipient: 'reporting_unit'
+      }]
+    }],
+    schedules: [{
+      name: 'ANC Reminders LMP',
+      summary: '',
+      description: '',
+      start_from: 'lmp_date',
+      messages: [
+        {
+          message: [{
+            content: 'Visit 1 reminder for {{patient_name}}',
+            locale: 'en'
+          }],
+          group: 1,
+          offset: '15 weeks',
+          send_day: 'monday',
+          send_time: '09:00',
+          recipient: 'reporting_unit'
+        },
+        {
+          message: [{
+            content: 'Visit 2 reminder for {{patient_name}}',
+            locale: 'en'
+          }],
+          group: 2,
+          offset: '20 weeks',
+          send_day: 'monday',
+          send_time: '09:00',
+          recipient: 'reporting_unit'
+        }
+      ]
+    }]
+  };
+  const savedUuids = [];
+
+  const submit = body => {
+    const content = JSON.stringify(body);
+    return utils.request({
+      method: 'POST',
+      path: '/api/sms',
+      body: content,
+      headers: {
+        'Content-Type': 'application/json',
+        'Content-Length': content.length
+      }
+    });
+  };
+
+  // refresh after changing settings so the new settings are loaded
+  const refresh = () => {
+    browser.driver.navigate().refresh();
+    return browser.wait(() => browser.isElementPresent(by.id('reports-tab')), 10000);
+  };
+
+  describe('submits new sms messages', () => {
+
+    beforeEach(done => {
+      browser.ignoreSynchronization = true;
+      const body = {
+        messages: [{
+          from: PHONE,
+          content: 'R Siobhan 12',
+          id: 'a'
+        }]
+      };
+      utils.updateSettings(CONFIG)
+        .then(() => protractor.promise.all(DOCS.map(utils.saveDoc)))
+        .then(results => results.forEach(result => savedUuids.push(result.id)))
+        .then(() => submit(body))
+        .then(() => refresh)
+        .then(done)
+        .catch(done);
```

Comment:
There's a proposal for one. I don't think it would work here because the `.catch(done)` passes the given `err` to `done` and fails the test, whereas I assume `finally` wouldn't have an `err`.